tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -78,22 +78,25 @@ try {
       // pointer we just got.
    }
 
-   // The "fast" option is present if and only if we don't want to change compression.
-   const int compression =
-      mergeInfo->fOptions.Contains("fast") ? kUnknownCompressionSettings : outFile->GetCompressionSettings();
-
-   RNTupleWriteOptions writeOpts;
-   writeOpts.SetUseBufferedWrite(false);
-   if (compression != kUnknownCompressionSettings)
-      writeOpts.SetCompression(compression);
-   auto destination = std::make_unique<RPageSinkFile>(ntupleName, *outFile, writeOpts);
-
-   // If we already have an existing RNTuple, copy over its descriptor to support incremental merging
-   if (outNTuple) {
-      auto source = RPageSourceFile::CreateFromAnchor(*outNTuple);
-      source->Attach();
-      auto desc = source->GetSharedDescriptorGuard();
-      destination->InitFromDescriptor(desc.GetRef());
+   const bool defaultComp = mergeInfo->fOptions.Contains("default_compression");
+   const bool firstSrcComp = mergeInfo->fOptions.Contains("first_source_compression");
+   if (defaultComp && firstSrcComp) {
+      // this should never happen through hadd, but a user may call RNTuple::Merge() from custom code...
+      Warning(
+         "RNTuple::Merge",
+         "Passed both options \"default_compression\" and \"first_source_compression\": only the latter will apply.");
+   }
+   int compression = kUnknownCompressionSettings;
+   if (firstSrcComp) {
+      // user passed -ff or -fk: use the same compression as the first RNTuple we find in the sources.
+      // (do nothing here, the compression will be fetched below)
+   } else if (!defaultComp) {
+      // compression was explicitly passed by the user: use it.
+      compression = outFile->GetCompressionSettings();
+   } else {
+      // user passed no compression-related options: use default
+      compression = RCompressionSetting::EDefaults::kUseGeneralPurpose;
+      Info("RNTuple::Merge", "Using the default compression: %d", compression);
    }
```

Comment:
Can you remind me what is the default if I just do `hadd out.root in1.root in2.root`? From a users perspective, I would not expect this to change the compression / recompress, but the code seems to suggest that I have to pass `-ff` or `-fk` to get "fast" merging?